osu.Server.Spectator.Tests/MultiplayerFlowTests.cs

```diff
@@ -539,6 +551,42 @@ public async void NotReadyUsersDontGetLoadRequest()
 
         #endregion
 
+        #region Beatmap Availability
+
+        [Fact]
+        public async Task ClientCantChangeAvailabilityWhenNotJoinedRoom()
+        {
+            await Assert.ThrowsAsync<NotJoinedRoomException>(() => hub.ChangeBeatmapAvailability(BeatmapAvailability.Importing()));
+        }
+
+        [Fact]
+        public async Task OnlyClientsInSameRoomReceiveAvailabilityChange()
+        {
+            var room = await hub.JoinRoom(room_id);
+
+            setUserContext(mockContextUser2);
+            var room2 = await hub.JoinRoom(room_id_2);
+
+            var user1Availability = BeatmapAvailability.Importing();
+            var user2Availability = BeatmapAvailability.Downloading(0.5);
+
+            setUserContext(mockContextUser1);
+            await hub.ChangeBeatmapAvailability(user1Availability);
+            Assert.True(room.Users.Single().BeatmapAvailability.Equals(user1Availability));
+
+            setUserContext(mockContextUser2);
+            await hub.ChangeBeatmapAvailability(user2Availability);
```

Comment:
You aren't covering the case that this is triggered twice, where the second fire should not send a message to other users in the room. This would be visible if you run code coverage and check which lines are not covered by your tests.